This is just a record of how the image IDs for skull stripped MRI images was matched to their respective gene expression data. This will be used later in multi-modal training experiments.

First from the the .csv file for the whole  MRI database is downloaded that has all the detailed summary of all the MRI samples in ADNI. This file present in current directory is: 'wholedata.csv'

In [5]:
import os
whole_csv_path = '/Users/subashkhanal/Desktop/MLforAlzheimers/data/ADNI/'
combined_gene_expr_path = '/Users/subashkhanal/Desktop/MLforAlzheimers/data/ADNI/genetics/data_prep'
os.listdir(whole_csv_path)

['my_mri_gene.csv',
 'ADNI_Imaging_Gene_Expression_MMSE_Combined.csv',
 '.DS_Store',
 'skull_stripped_MRI',
 'wholedata.csv',
 'final_skull_stripped_mri_with_gene_expression.csv',
 'skull_stripped.csv',
 'skull_stripped_metadata',
 'genetics']

The skull stripped MRIs have the Description field as 'Spatially Normalized, Masked and N3 corrected T1 image'. So..

In [3]:
import pandas as pd
sub = []
whole_df = pd.read_csv(os.path.join(whole_csv_path,'wholedata.csv'))
for i in range(len(whole_df)):
    if 'Spatially Normalized, Masked and N3 corrected T1 image' in whole_df.iloc[i]['Description']:
        sub.append(whole_df.iloc[i]['Subject ID'])
len(sub)

1026

In [7]:
gene_df = pd.read_csv(os.path.join(combined_gene_expr_path,'ADNI_Imaging_Gene_Expression_MMSE_Combined.csv'),low_memory=False)

In [12]:
subjects_gene = gene_df.columns[1:]
len(subjects_gene)

744

In [15]:
common_subs = []
for s in set(sub):
    if s  in list(subjects_gene):
        common_subs.append(s)
len(common_subs)

134

This shows only 134 different subjects have both gene expression and skull stripped MRI in ADNI. We need to work on getting our own pre-processing pipeline for skull-stripping!!

In [17]:
txt = ''
for s in common_subs:
    txt = txt+s+','
txt

'109_S_2278,003_S_4441,003_S_1074,098_S_4050,052_S_1352,126_S_4507,005_S_2390,016_S_2284,029_S_4307,094_S_2238,098_S_4506,029_S_4279,098_S_4018,127_S_4645,016_S_4009,021_S_4335,109_S_4455,099_S_4498,127_S_4210,007_S_4467,135_S_4309,094_S_2367,129_S_2332,021_S_4254,016_S_2031,021_S_4419,099_S_4480,021_S_4421,127_S_4148,003_S_4152,003_S_4350,007_S_2394,052_S_4626,129_S_4287,127_S_2234,131_S_0123,016_S_4097,094_S_4434,003_S_4081,109_S_4531,029_S_4385,129_S_4371,005_S_4707,099_S_2146,099_S_4565,003_S_4555,029_S_4384,007_S_4620,099_S_4463,005_S_4168,016_S_4353,016_S_4584,007_S_4272,021_S_2125,029_S_4327,003_S_0908,098_S_4215,003_S_2374,135_S_4356,021_S_4558,126_S_4458,057_S_2398,126_S_4494,057_S_0934,003_S_4373,021_S_2124,098_S_4275,126_S_2407,094_S_2216,127_S_4198,016_S_4591,094_S_4089,094_S_4234,003_S_4136,099_S_2063,094_S_4162,005_S_4185,094_S_2201,129_S_4220,129_S_4369,135_S_4281,027_S_2245,021_S_4245,021_S_2142,029_S_2395,094_S_4560,021_S_4276,094_S_4503,007_S_4387,099_S_4205,003_S_435

Using these subjects download the detailed .csv file for the MRIs from ADNI database. The downloaded .csv is named here as 'my_mri_gene.csv'.

In [26]:
df_final = pd.read_csv(os.path.join(whole_csv_path,'my_mri_gene.csv'))

In [27]:
len(set(df_final['Subject ID']))

134

In [28]:
sub = []
for i in range(len(df_final)):
    if 'Spatially Normalized, Masked and N3 corrected T1 image' in df_final.iloc[i]['Description']:
        sub.append(i)

In [29]:
from collections import Counter
df_final = df_final.iloc[sub]
Counter(df_final['Research Group'])

Counter({'MCI': 4, 'EMCI': 246, 'CN': 171, 'AD': 60, 'LMCI': 70})

In [32]:
len(df_final)

551

In [33]:
txt = ''
for i in df_final['Image ID']:
    txt = txt+str(i)+','
txt

'420849,537823,298789,356736,537831,298790,397037,298791,298961,356737,298792,298901,413116,383267,298794,369559,298795,298902,334838,413125,420865,298796,366706,369568,420866,698586,298797,397000,334806,369569,420867,698587,298798,298799,356696,383268,420868,334748,356697,437848,786086,298962,298801,298903,334860,397038,537846,298802,334839,356738,397039,614179,298803,298904,334840,356739,397040,614180,334750,334830,356698,383269,298963,298804,298905,298964,298805,298906,299000,413172,298965,298806,298907,334861,397041,763052,298807,298908,334841,356740,413156,298808,298909,366708,366732,437849,763053,298809,334811,334855,366722,420869,298810,356699,366723,298811,334812,356700,420870,698588,298812,356656,356701,383270,420871,334751,356702,369570,437850,763054,334752,334813,356703,383271,449736,763055,334753,356657,356704,383272,437851,763056,298967,298814,298911,299001,356761,397052,298815,298912,298968,298816,298913,298974,334862,383295,298817,298914,298975,298818,298915,298976,35674

# Using these image ids skull stripped MRIs are downloaded for which there is availability of gene expression profile data.